pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog"
+	internalcache "k8s.io/kubernetes/pkg/scheduler/internal/cache"
+)
+
+// Framework is the component responsible for initializing and
+type Framework struct {
+	registry       Registry
+	pluginContext  *PluginContext
```

Comment:
I'm confused why `PluginContext` is a top level framework object. Would it be more accurate to name this `FrameworkHandle`?

Reply:
I thought framework would manage ephemeral pluginContexts for each active scheduling cycle.